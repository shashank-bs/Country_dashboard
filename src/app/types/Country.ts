export interface Country {
    name: {
      common: string;
      official:string;
    };
    population: number;
    region: string;
    capital: string[];
    flags: {
      svg: string;
    };
    currencies: Record<string, { name: string }>;
    languages: Record<string, string>;
    timezones: string[];
    cca2:string;
  }
  