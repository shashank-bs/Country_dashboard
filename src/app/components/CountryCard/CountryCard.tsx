// app/components/CountryCard.tsx
import { Card, CardContent, Typography } from "@mui/material";
import Image from "next/image";
import Link from "next/link";
import { Country } from "../../types/Country";

export interface CountryCardProps {
  country: Country;
}

const CountryCard: React.FC<CountryCardProps> = ({ country }) => {
  // URL encode the country name for the route
  const countryNameUrl = encodeURIComponent(country.name.common.toLowerCase());

  return (
    <Link href={`/${countryNameUrl}`}>
      <Card
        sx={{ maxWidth: 345, cursor: "pointer" }}
        data-testid="country-card"
        variant="elevation"
      >
        <div style={{ textAlign: "center" }}>
          <Image
            src={country.flags.svg}
            alt={`Flag of ${country.name.common}`}
            width={150}
            height={100}
            style={{ objectFit: "cover" }}
            placeholder="blur"
            blurDataURL={country.flags.svg}
          />

          <CardContent>
            <Typography gutterBottom variant="h5" component="div">
              {country.name.common}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              Capital: {country.capital?.[0] || "N/A"}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              Population: {country.population.toLocaleString()}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              Region: {country.region}
            </Typography>
          </CardContent>
        </div>
      </Card>
    </Link>
  );
};

export default CountryCard;
