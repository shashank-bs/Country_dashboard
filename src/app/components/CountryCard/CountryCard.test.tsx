import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CountryCard from "./CountryCard";
import { Country } from "../../types/Country";

describe("CountryCard Component", () => {
  const mockCountry: Country = {
    name: {
      common: "United States",
      official: "",
    },
    flags: { svg: "https://example.com/flag.svg" },
    capital: ["Washington, D.C."],
    population: 331000000,
    region: "Americas",
    currencies: {},
    languages: {},
    timezones: [],
    cca2: "",
  };

  test("renders country name correctly", () => {
    const { getByText } = render(<CountryCard country={mockCountry} />);
    expect(getByText("United States")).toBeInTheDocument();
  });

  test("renders country capital correctly", () => {
    const { getByText } = render(<CountryCard country={mockCountry} />);
    expect(getByText("Capital: Washington, D.C.")).toBeInTheDocument();
  });

  test("renders country region correctly", () => {
    const { getByText } = render(<CountryCard country={mockCountry} />);
    expect(getByText("Region: Americas")).toBeInTheDocument();
  });

  test("renders country flag correctly", () => {
    const { getByAltText } = render(<CountryCard country={mockCountry} />);
    expect(getByAltText("Flag of United States")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<CountryCard country={mockCountry} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
