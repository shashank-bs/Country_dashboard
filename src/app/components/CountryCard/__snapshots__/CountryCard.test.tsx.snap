// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard Component matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/united%20states"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rzkyto-MuiPaper-root-MuiCard-root"
      data-testid="country-card"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        style="text-align: center;"
      >
        <img
          alt="Flag of United States"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="https://example.com/flag.svg"
          style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="150"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-13y79c5-MuiTypography-root"
          >
            United States
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
          >
            Capital: Washington, D.C.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
          >
            Population: 33,10,00,000
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 css-atovpb-MuiTypography-root"
          >
            Region: Americas
          </p>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
